src/codeu/chat/client/core/Context.java

```diff
@@ -19,13 +19,14 @@
 import java.util.Collection;
 
 import codeu.chat.common.BasicView;
+import codeu.chat.common.ServerInfo;
 import codeu.chat.common.User;
 import codeu.chat.util.Uuid;
 import codeu.chat.util.connections.ConnectionSource;
 
 public final class Context {
 
-  private final BasicView view;
+  private final View view;
```

Comment:
You have to define the method in the interface "BasicView", and you already implemented it in View. Now you are making this class dependent of a specific implementation(View). The View Class is responsible for the network calls, consider that you want to implement a different ViewV2 class that uses a different code to do the network calls, now this class won't work because you make it dependent on a specific implementation.